### PR TITLE
Ensured items are not required for a payment request

### PR DESCRIPTION
**Pull request by joel-maizonnier**
## Description of Problem

In order to complete a payment request with discount items, it's not possible to provide a discount item (negative value or a separate discount item).

Therefore in order to complete a payment request with a discount, no items can be provided, but validation rule require at least one item to be provided.

## Description of Changes

* Allowed a payment request to be made without providing any items for the request
* The item validation rules has been updated to check if a negative value has been provided, for a items price

### CODE REVIEW

File: tests/unit/Request/PaymentRequestTest.php
Comment (tommi-martin) [inline tests/unit/Request/PaymentRequestTest.php:25]: Typo in the word test.

File: tests/unit/Request/PaymentRequestTest.php
Comment (tommi-martin) [inline tests/unit/Request/PaymentRequestTest.php:10]: In regards to the unit tests I would consider increasing the coverage of your tests by adding negative validation. Essentially creating assertions about when and how the validation will throw out an exception. This will allow you to lock down validation types using tests. And if someone changes the criteria or makes a mistake in validation that should've thrown an exception. The tests will catch it.

File: tests/unit/Request/PaymentRequestTest.php
Comment (joel-maizonnier) [reply tests/unit/Request/PaymentRequestTest.php]: Done

File: tests/unit/Request/PaymentRequestTest.php
Comment (joel-maizonnier) [reply tests/unit/Request/PaymentRequestTest.php:10]: Good point, I have added this test to the existing items test. I've also noticed that testing for the abstract PaymentRequest was also missing so I've added a new test to cover it's validation messages.

**Review by tommi-martin — APPROVED** [review]
Looks good to me

**Comment by joel-maizonnier** [comment]
> Looks good to me

Great :+1: Can these changes be merged into master?

**Comment by tommi-martin** [comment]
Before you do, please wait for a review from 
@eerohakio and @loueranta-paytrail  as they're the ones with write access to the original SDK repository.

**Review by eerohakio — APPROVED** [review]
Looks good to me, good work 👍

**Review by loueranta-paytrail — APPROVED** [review]
LGTM! 👍

**Comment by joel-maizonnier** [comment]
Can new version tag be added to master so I can tell composer to use a certain version? I'm currently having to use dev-master :laughing:

**Pull request merged**
